maro/simulator/scenarios/supply_chain/business_engine.py

```diff
@@ -129,13 +209,104 @@ def _build_world(self) -> None:
 
         self.world.build(conf, self.calc_max_snapshots(), self._max_tick)
 
+    def _build_dag(self) -> None:
```

Comment:
I'm wondering whether the dag and the task scheduler are necessary. 

Put the ordered logics into a function naming like `step_after_action_received`, and let `on_action_received` to be like:
```
def on_action_recieved(self, event):
    self._action_by_unit = ...
    self.step_after_action_received()

def step_after_action_received(self, tick):
    for consumer_unit in self._consumer_units:
        if consumer_unit.id in self.actions_by_unit:
            ...

    for distribution_unit in ...:
        try_schedule..

    for distribution_unit in:
        handle_arrival...

    for manufacture_unit in...
        if manufacture_unit.id in self.actions_by_unit:
            ...
        execute_manufacture...

        
```

Would it be clearer?